Src/library/docking/implementation/docking/sd_docking_manager_query.e

```diff
@@ -13,28 +13,38 @@ inherit
 
 	SD_DOCKING_MANAGER_HOLDER
 
+create
+	make
+
+feature {NONE} -- Creation
+
+	make (a_docking_manager: SD_DOCKING_MANAGER)
+			-- Associate new object with `a_docking_manager'.
+		do
+			docking_manager := a_docking_manager
+		end
+
 feature -- Querys
 
 	auto_hide_panel (a_direction: INTEGER): SD_AUTO_HIDE_PANEL
 			-- Auto hide panel at `a_direction'
 		require
 			a_direction_valid: a_direction = {SD_ENUMERATION}.top or a_direction = {SD_ENUMERATION}.bottom
 				or a_direction = {SD_ENUMERATION}.left or a_direction = {SD_ENUMERATION}.right
-		local
-			l_result: detachable like auto_hide_panel
-		do
-			if a_direction = {SD_ENUMERATION}.bottom then
-				l_result := docking_manager.internal_auto_hide_panel_bottom
-			elseif a_direction = {SD_ENUMERATION}.top then
-				l_result := docking_manager.internal_auto_hide_panel_top
-			elseif a_direction = {SD_ENUMERATION}.left then
-				l_result := docking_manager.internal_auto_hide_panel_left
-			elseif a_direction = {SD_ENUMERATION}.right then
-				l_result := docking_manager.internal_auto_hide_panel_right
+		do
+			inspect a_direction
+			when {SD_ENUMERATION}.bottom then
+				Result := docking_manager.internal_auto_hide_panel_bottom
+			when {SD_ENUMERATION}.top then
+				Result := docking_manager.internal_auto_hide_panel_top
+			when {SD_ENUMERATION}.left then
+				Result := docking_manager.internal_auto_hide_panel_left
+			when {SD_ENUMERATION}.right then
+				Result := docking_manager.internal_auto_hide_panel_right
+			end
+			if not attached Result then
```

Comment:
No need for the check since there is no else clause in the inspect, we are guaranteed that Result is attached if all the above assignments have this post-condition.